packages/mui-system/src/cssVars/createCssVarsProvider.test.js

```diff
@@ -237,6 +238,66 @@ describe('createCssVarsProvider', () => {
       expect(screen.getByText('var(--palette-primary)')).not.to.equal(null);
       expect(screen.getByText('var(--palette-grey)')).not.to.equal(null);
     });
+
+    it('disable all css transitions when switching bewteen modes given `disableTransitionOnChange` is true', () => {
+      const { CssVarsProvider, useColorScheme } = createCssVarsProvider({
+        theme: {
+          colorSchemes: { light: {}, dark: {} },
+        },
+        defaultColorScheme: 'light',
+        disableTransitionOnChange: true,
+      });
+      const Consumer = () => {
+        const { mode, setMode } = useColorScheme();
+        return (
+          <div>
+            <div data-testid="current-mode">{mode}</div>
+            <button onClick={() => setMode('dark')}>change to dark</button>;
+          </div>
+        );
+      };
+      render(
+        <CssVarsProvider>
+          <Consumer />
+        </CssVarsProvider>,
+      );
+
+      fireEvent.click(screen.getByRole('button', { name: 'change to dark' }));
+      expect(document.head.children[document.head.children.length - 1].textContent).to.equal(
+        DISABLE_CSS_TRANSITION,
+      );
```

Comment:
The test will be tricky because the `disableTransition` function contains `setTimeout`.

Your test should be like this:
1. render
2. expect that `DISABLE_CSS_TRANSITION` is not on head (because it should only appear on change, not on mount)
3. change color scheme or mode
4. assert that `DISABLE_CSS_TRANSITION` is on head
5. run timer
6. assert that `DISABLE_CSS_TRANSITION` is NOT on head (make sure that we get rid of it.